eslint.config.mjs

```diff
@@ -857,15 +731,63 @@ export default typescript.config([
    *   ...myPlugin.configs.recommended,
    *   rules: {
    *     ...myPlugin.configs.recommended.rules,
-   *     ['the-rule']: 'warning',
+   *     ['the-rule']: 'warn',
    *   }
    * },
    * Finally, once all warnings are fixed, update from 'warning' to 'error', or
    * remove the override and rely on the recommended rules again.
    */
   {
-    name: 'import/recommended',
+    name: 'import',
     ...importPlugin.flatConfigs.recommended,
+    rules: {
+      // We override and specifically set all the rules that are in the
+      // recommended, react, and typescript rulesets. No need to bring those in
+
+      // From the recommended ruleset:
+      // https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/export.md
+      'import/export': 'error',
+
+      // 5 rules not recommended to be enabled with typescript-eslint
+      // https://typescript-eslint.io/troubleshooting/typed-linting/performance/#slow-eslint-rules
+      'import/named': 'off',
+      'import/namespace': 'off',
+      'import/default': 'off',
+      'import/no-named-as-default-member': 'off',
+      'import/no-unresolved': 'off',
+
+      // Require a newline after the last import/require in a group
+      // https://github.com/benmosher/eslint-plugin-import/blob/main/docs/rules/newline-after-import.md
+      'import/newline-after-import': 'error', // TODO: should be a prettier thing?
```

Comment:
I think we can turn this off, prettier will add the newline

Reply:
oh yeah, i'll test to double-check

Reply:
https://prettier.io/docs/en/rationale.html#empty-lines

prettier will collapse too many lines, but won't add more.

Reply:
i've added a comment. 

I think we should followup and set it to be: 
```
      'import/newline-after-import': [
        'error',
        {count: 1, exactCount: true, considerComments: true},
      ],
```

there are ~50 places that'll get auto fixed up. but i'll shove that into it's own PR
```suggestion
      'import/newline-after-import': 'error', // TODO: should be a prettier thing?
```